src/app/services/downloads.service.ts

```diff
@@ -0,0 +1,64 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, timer } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { ExportResult, QhanaBackendService } from './qhana-backend.service';
+
+interface ExportCache {
+    exportId: number;
+    status: "SUCCESS" | "FAILURE" | "PENDING";
+}
+
+@Injectable({
+    providedIn: 'root'
+})
+export class DownloadsService {
+    private exportsCache: ExportCache[] | null = null;
+    private exportList: BehaviorSubject<ExportResult[] | null> = new BehaviorSubject<ExportResult[] | null>(null);
+    private downloadsCounter: BehaviorSubject<number> = new BehaviorSubject(0);
+
+    constructor(private backend: QhanaBackendService) {
+        timer(0, 2000).subscribe(() => {
+            var resp = this.backend.getExportList().pipe(
+                map(resp => {
+                    this.updateExports(resp);
+                    this.exportList.next(resp);
+                    return resp;
+                })
+            )
+            // hacky, but need pseudosubscription to start polling
+            resp.subscribe(() => console.log());
+        })
+    }
+
+    getExportList(): BehaviorSubject<ExportResult[] | null> {
+        return this.exportList;
```

Comment:
do not expose your subjects directly. instead expose observables. You can allow interaction with the subjects through methods defined on the service.